src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,426 @@
+        app_manager_.GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *ui_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        hmi_apis::FunctionID::UI_SetGlobalProperties;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreateVRMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_.ManageHMICommand(ui_global_properties);
+  }
+}
+
+void HelpPromptManagerImpl::SendBothRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  SendTTSRequest();
+  SendUIRequest();
+}
+
+void HelpPromptManagerImpl::SendRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+
+  switch (sending_type_) {
+    case SendingType::kSendHelpPromt:
+      SendTTSRequest();
+      break;
+    case SendingType::kSendVRHelp:
+      SendUIRequest();
+      break;
+    case SendingType::kSendBoth:
+      SendBothRequests();
+      break;
+    default:
+      LOG4CXX_DEBUG(logger_,
+                    "SendingType:" << (unsigned)sending_type_
+                                   << " request not sending");
+      return;
+  }
+}
+
+void HelpPromptManagerImpl::CreatePromtMsg(
+    smart_objects::SmartObject& out_msg_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  out_msg_params[strings::help_prompt] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  VRCommandsMap::const_iterator it = vr_commands_.begin();
+  uint32_t index = 0;
+  for (; it != vr_commands_.end(); ++it) {
+    for (unsigned i = 0; i < it->second->length(); ++i) {
+      smart_objects::SmartObject item(smart_objects::SmartType_Map);
+
+      item[strings::text] = it->second->getElement(i).asString();
+      item[strings::type] = mobile_apis::SpeechCapabilities::SC_TEXT;
+
+      out_msg_params[strings::help_prompt][index++] = item;
+    }
+  }
+  app_.set_help_prompt(out_msg_params[strings::help_prompt]);
+}
+
+void HelpPromptManagerImpl::CreateVRMsg(
+    smart_objects::SmartObject& out_msg_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (vr_commands_.empty()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  if (false == out_msg_params.keyExists(strings::vr_help_title)) {
+    if (app_.vr_help_title()) {
+      out_msg_params[strings::vr_help_title] = (*app_.vr_help_title());
+    } else {
+      out_msg_params[strings::vr_help_title] = app_.name();
+    }
+  }
+  out_msg_params[strings::vr_help] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  VRCommandsMap::const_iterator it = vr_commands_.begin();
+  uint32_t index = 0;
+  for (; it != vr_commands_.end(); ++it) {
+    for (unsigned i = 0; i < it->second->length(); ++i) {
```

Comment:
unsigned i  --> std::size_t i

Reply:
Thanks, fixed in https://github.com/smartdevicelink/sdl_core/pull/2186/commits/1de0eb183562592396687a028c121176c53c0aeb